pkg/stanza/adapter/receiver_test.go

```diff
@@ -178,7 +174,11 @@ func BenchmarkReadLine(b *testing.B) {
 	require.NoError(b, yaml.Unmarshal([]byte(pipelineYaml), &operatorCfgs))
 
 	set := componenttest.NewNopTelemetrySettings()
-	emitter := helper.NewLogEmitter(set)
+	emitter := helper.NewLogEmitter(set, func(_ context.Context, entries []*entry.Entry) {
```

Comment:
That's a good point, and adapting the benchmarks first in a separate PR first  makes sense to me. I will start looking into this today

Reply:
Alright, I just created a first draft PR for the new test: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35833 - @djaglowski feel free to have a look if this is what you had in mind

Reply:
I left one comment about the structure of the entries we are converting, but otherwise the new benchmark looks good. Looking forward to seeing if this shows a clearer diff of the changes in this PR.

Reply:
With https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35833 merged, is this now resolved?

Reply:
Just had another look and I think it would make sense to also adapt this to work similarly to the newly introduced benchmark test, so I'll look into that today